app/queues/dns/add-record-flow.server.ts

```diff
@@ -0,0 +1,103 @@
+import { FlowProducer } from 'bullmq';
+import invariant from 'tiny-invariant';
+import { redis } from '~/lib/redis.server';
+import { buildDomain } from '~/utils';
+import { dnsUpdateQueueName } from './workers/dns-update-worker.server';
+import { checkDnsStatusQueueName } from './workers/poll-dns-status-worker.server';
+import { syncDbStatusQueueName } from './workers/sync-db-status-worker.server';
+import { createUserRecord } from '~/models/record.server';
+
+import type { FlowJob } from 'bullmq';
+import type { Record } from '@prisma/client';
+import type { DnsUpdaterData } from './workers/dns-update-worker.server';
+import type { DbRecordSynchronizerData } from './workers/sync-db-status-worker.server';
+
+export type FqdnOrSubdomain = { subdomain: string } | { fqdn: string };
+interface BaseAddDnsRequestData {
+  username: Record['username'];
+  type: Record['type'];
+  value: Record['value'];
+  description?: Record['description'];
+  course?: Record['course'];
+  ports?: Record['ports'];
+}
+export type AddDnsRequestData = BaseAddDnsRequestData & FqdnOrSubdomain;
+
+export enum WorkType {
+  create = 'create',
+  update = 'update',
+  delete = 'delete',
+}
+
+const flowProducer = new FlowProducer({ connection: redis });
+
+export const addDnsRequest = async (data: AddDnsRequestData) => {
+  const { username, type, value } = data;
+
+  invariant('subdomain' in data, 'expected subdomain vs. fqdn');
```

Comment:
If we are throwing an exception if fqdn is provided, then there is no point in accepting it in the first place.

If the whole system gravitates toward only working with a subdomain, then let's not accept fqdn here, I'll alter the dns worker send a subdomain instead

Reply:
I'm not sure how much I understand the whole system. But I think there would not be many places using `fqdn`.